### PR TITLE
feat: add events to history manager

### PR DESCRIPTION
**Pull request by jcdrouin21**
# Summary of Changes

This PR aims to add the majority of events that we might want to log in the history. It allows each wallet to have control over the events it wants to add in the history.
Exemple of some entries:
image

# Screenshots, videos, or gifs

Replace this text with embedded media for UI changes if they are included in this PR. If there are none, simply enter N/A

# Breaking change guide

Replace this text with any breaking changes included in this PR along with how to address them in downstream projects. If there are none, simply enter N/A

# Related Issues

Replace this text with issue #'s that are relevant to this PR. If there are none, simply enter N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this)
- [x] If applicable, screenshots, gifs, or video are included for UI changes
- [x] If applicable, breaking changes are described above along with how to address them
- [x] Updated documentation as needed for changed code and new or modified features
- [ ] Added sufficient [tests](../packages/legacy/core/__tests__/) so that overall code coverage is not reduced

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated checks have passed

### CODE REVIEW

**Comment by MosCD3** [comment]
Very nice work but this has to go after #1333

**Review by MosCD3 — CHANGES_REQUESTED** [review]
Can you rebase and fix accordingly

**Comment by fc-santos** [comment]
@MosCD3 fixed the events to use historyEnabled flag from container instead of store.preferences.useHistoryCapability and rebased with main. Can you review it ? Thanks!

File: packages/legacy/core/App/index.ts [hunk 1]
```diff
@@ -141,6 +141,7 @@ export {
   AttachTourStep,
   TourBox,
   defaultConfig,
+  defaultHistoryEventsLogger,
```
Comment (MosCD3) [inline packages/legacy/core/App/index.ts:146]: Can you also export `HistoryEventsLoggerConfig`

Comment (fc-santos) [reply packages/legacy/core/App/index.ts:145]: @MosCD3 done!

**Comment by fc-santos** [comment]
@MosCD3 done!

**Review by MosCD3 — APPROVED** [review]

**Comment by MosCD3** [comment]
Please rebase and merge

**Pull request merged**